### PR TITLE
fix/requires-confirmation-with-payment

### LINKED ISSUE #5542: [CAL-416] Bookings are bypassing required confirmation and worflows/webhooks are not firing
Found a bug? Please fill out the sections below. 👍

### Summary

I had a booking submitted around 4 pm EST, which requires both payment and confirmation. The booking was immediately placed in "upcoming" but did not appear under "unconfirmed." It also did not give me the option of confirming the appointment. I checked the settings and made sure that it was set up to require a confirmation from me due to the nature of the appointment. All settings appear configured properly.

Additionally, any triggers or webhooks that have been set up are not activating. I should have received two separate workflow notifications for that event, but I didn't receive anything. I went into the workflow to confirm it was attached to the event, that it was enabled, and then tested it. The test worked and I received my notification, but it does not happen without manually triggering a test. The webhook did not fire at all, nor did my Zapier integration.

### Steps to Reproduce

1. Configure an appointment to require confirmation and set up workflows that are triggered for booking
2. Tested the workflow triggers and they worked without issue
3. Tested webhook configuration and pulled test data into Zapier without issue
4. When a booking is done organically, it is automatically confirmed and/or bypasses needing confirmation entirely
5. The appointment is added to upcoming, with no option to confirm the booking, just to cancel or reschedule
6. No notifications or triggers alerted me to the booking, rendering all automation non-existent.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

1. When an event is configured to require booking confirmation, then the event is added as unconfirmed with the option of confirming the appointment or rejecting it.
2. When the trigger for a new booking occurs, all enabled workflows and webhooks should engage without manual interaction.

### Technical details

N/A

> From [SyncLinear.com](https://synclinear.com)

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Enables a booking to stay unconfirmed after payment its made
- Add badge in booking list to display if payment its true

Fixes #5542 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [ ] Create event that requires payment and requires confirmation
- [ ] Setup stripe webhooks
- [ ] Go to booking that event, make the payment. Booking should be unconfirmed but flagged as paid

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -284,6 +284,11 @@ function BookingListItem(booking: BookingItemProps) {
                 {t("pending_payment")}
               </Badge>
             )}
+            {booking.paid && (
+              <Badge className="ltr:mr-2 rtl:ml-2" variant="green">
+                {t("paid")}
+              </Badge>
+            )}
```
Comment (alannnc) [inline apps/web/components/booking/BookingListItem.tsx:291]: same as other badges

File: packages/features/ee/payments/api/webhook.ts [hunk 2]
```diff
@@ -65,7 +65,10 @@ async function handlePaymentSuccess(event: Stripe.Event) {
 
   if (!booking) throw new HttpCode({ statusCode: 204, message: "No booking found" });
 
-  const eventTypeSelect = Prisma.validator<Prisma.EventTypeSelect>()({ recurringEvent: true });
+  const eventTypeSelect = Prisma.validator<Prisma.EventTypeSelect>()({
+    recurringEvent: true,
+    requiresConfirmation: true,
```
Comment (alannnc) [inline packages/features/ee/payments/api/webhook.ts:70]: now reading required field

File: packages/features/ee/payments/api/webhook.ts [hunk 3]
```diff
@@ -130,6 +133,10 @@ async function handlePaymentSuccess(event: Stripe.Event) {
     bookingData.references = { create: scheduleResult.referencesToCreate };
   }
 
+  if (eventTypeRaw?.requiresConfirmation) {
+    delete bookingData.status;
```
Comment (alannnc) [inline packages/features/ee/payments/api/webhook.ts:137]: booking status its set as SUCESS on line 123/126

File: packages/trpc/server/routers/viewer/eventTypes.tsx [hunk 4]
```diff
@@ -504,7 +504,8 @@ export const eventTypesRouter = router({
       };
     }
 
-    if (input?.price) {
+    if (input?.price || input.metadata?.apps?.stripe?.price) {
+      data.price = input.price || input.metadata?.apps?.stripe?.price;
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/eventTypes.tsx:508]: Just realised that after v2 we are loosing eventType price attribute from being updated so we read input price or stripe metadata price.

**Review by zomars — APPROVED** [review]

**Review by zomars — APPROVED** [review]
Ship it

**Pull request merged**